fellowship/rest_tester.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 Nokia
+# Licensed under the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import json
+import requests
+
+from jsonschema import validate, ValidationError
+
+from .contract_renderer import ContractRenderer
+from .reporter import Reporter
+
+
+class RestTester:
+    """Validates Rest endpoints based on contracts
+    Attributes:
+        contract_dir (str): Path to directory of contracts to validate
+        contract_renderer (object): Contract render module used to fill the Jinja2
+            template of the contracts.
+    """
+    _requests_arg_names = ['headers', 'data', 'params']
+
+    def __init__(self, contract_dir: str):
+        self.contract_dir = contract_dir
+        self.contract_renderer = ContractRenderer(contract_dir)
+        self._reporter = None
+
+    def make_requests_and_validates(self) -> None:
+        """

```

Comment:
In the documentation it would be good to indicate what exceptions are raised. Please consider raising rest_tester module specific exceptions for the known error cases. They would be easier for the user to handle. If the generic exception like 'SystemExit' is raised, then this is difficult to differentiate from other causes of 'SystemExit'.